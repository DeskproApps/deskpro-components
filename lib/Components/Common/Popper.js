'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Arrow = exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _popper = require('popper.js');

var _popper2 = _interopRequireDefault(_popper);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Portal = require('./Portal');

var _Portal2 = _interopRequireDefault(_Portal);

var _noop = require('../utils/noop');

var _noop2 = _interopRequireDefault(_noop);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * A wrapper around popper.js, a container which uses absolute positioning
 * to place elements relative to other elements.
 *
 * @see https://popper.js.org
 */
var Popper = (_temp = _class = function (_React$Component) {
  _inherits(Popper, _React$Component);

  function Popper(props) {
    _classCallCheck(this, Popper);

    var _this = _possibleConstructorReturn(this, (Popper.__proto__ || Object.getPrototypeOf(Popper)).call(this, props));

    _this.updatePopper = function () {
      var _this$props = _this.props,
          target = _this$props.target,
          placement = _this$props.placement,
          offsetX = _this$props.offsetX,
          offsetY = _this$props.offsetY,
          detached = _this$props.detached,
          eventsEnabled = _this$props.eventsEnabled,
          preventOverflow = _this$props.preventOverflow,
          _onCreate = _this$props.onCreate,
          _onUpdate = _this$props.onUpdate;

      if (preventOverflow && !eventsEnabled) {
        console.warn('`eventsEnabled` should be true when `preventOverflow` is true.');
      } else if (detached && !eventsEnabled) {
        console.warn('`eventsEnabled` should be true when `detached` is true.');
      }

      if (!_this.target) {
        if (typeof target === "function") {
          _this.target = target();
        } else if (typeof target === "string") {
          _this.target = document.getElementById(target);
        } else {
          _this.target = target;
        }
      }

      _this.destroyPopper();
      _this.popper = new _popper2.default(_this.target, _this.node, {
        placement: placement,
        eventsEnabled: eventsEnabled,
        modifiers: {
          offset: {
            offset: offsetX + ', ' + offsetY
          },
          preventOverflow: {
            enabled: !preventOverflow
          }
        },
        onCreate: function onCreate() {
          return _onCreate(_this);
        },
        onUpdate: function onUpdate() {
          return _onUpdate(_this);
        }
      });
    };

    _this.destroyPopper = function () {
      if (_this.popper) {
        _this.popper.destroy();
        _this.popper = null;
      }
    };

    _this.node = null;
    _this.target = null;
    return _this;
  }

  _createClass(Popper, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.updatePopper();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this.destroyPopper();
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      if (prevProps.opened !== this.props.opened) {
        this.props.opened ? this.props.onOpen(this) : this.props.onClose(this);
      }
      this.updatePopper();
      this.popper.scheduleUpdate();
    }

    /**
     * Initializes or updates the this.popper object.
     */


    /**
     * Destroys the internal Popper instance.
     */

  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          opened = _props.opened,
          arrow = _props.arrow,
          detached = _props.detached,
          children = _props.children,
          className = _props.className,
          elementProps = _objectWithoutProperties(_props, ['opened', 'arrow', 'detached', 'children', 'className']);

      if (!opened) {
        return _react2.default.createElement('div', null);
      }

      var popper = _react2.default.createElement(
        'div',
        _extends({ ref: function ref(_ref) {
            return _this2.node = _ref;
          }, className: (0, _classnames2.default)('dp-popper', className) }, elementProps),
        children,
        arrow ? _react2.default.createElement(Arrow, null) : null
      );
      return detached ? _react2.default.createElement(
        _Portal2.default,
        null,
        popper
      ) : popper;
    }
  }]);

  return Popper;
}(_react2.default.Component), _class.propTypes = {
  /**
   * CSS classes to apply to the element.
   */
  className: _propTypes2.default.string,
  /**
   * Popper is placed in reference to this element.
   */
  target: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]).isRequired,
  /**
   * Placement applied to popper.
   */
  placement: _propTypes2.default.oneOf(['auto', 'auto-start', 'auto-end', 'top', 'top-start', 'top-end', 'right', 'right-start', 'right-end', 'bottom', 'bottom-start', 'bottom-end', 'left', 'left-start', 'left-end']),
  /**
   * Shifts the popper on its X axis.
   */
  offsetX: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string]),
  /**
   * Shifts the popper on its Y axis.
   */
  offsetY: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string]),
  /**
   * Applied to the z-index style.
   */
  zIndex: _propTypes2.default.number,
  /**
   * Whether the children are displayed or not.
   */
  opened: _propTypes2.default.bool,
  /**
   * Whether to add an arrow to the popper.
   */
  arrow: _propTypes2.default.bool,
  /**
   * True to attach the children to document.body.
   */
  detached: _propTypes2.default.bool,
  /**
   * Add resize/scroll events and start recalculating position of
   * the popper element when they are triggered.
   */
  eventsEnabled: _propTypes2.default.bool,
  /**
   * Prevents the popper from being positioned outside the boundary.
   */
  preventOverflow: _propTypes2.default.bool,
  /**
   * Called when the popper is created.
   */
  onCreate: _propTypes2.default.func,
  /**
   * Called when the popper is updated, this callback is not called
   * on the initialization/creation of the popper, but only on subsequent updates.
   */
  onUpdate: _propTypes2.default.func,
  /**
   * Called when the popper is opened.
   */
  onOpen: _propTypes2.default.func,
  /**
   * Called when the popper is closed.
   */
  onClose: _propTypes2.default.func
}, _class.defaultProps = {
  placement: 'auto',
  offsetX: '0px',
  offsetY: '0px',
  zIndex: 0,
  arrow: true,
  detached: false,
  opened: false,
  eventsEnabled: false,
  preventOverflow: false,
  onCreate: _noop2.default,
  onUpdate: _noop2.default,
  onOpen: _noop2.default,
  onClose: _noop2.default
}, _temp);

/**
 * Arrow added to Popper containers.
 *
 * @returns {XML}
 */

exports.default = Popper;
var Arrow = function Arrow(_ref2) {
  var className = _ref2.className,
      elementProps = _objectWithoutProperties(_ref2, ['className']);

  return _react2.default.createElement('div', _extends({
    className: (0, _classnames2.default)('dp-popper__arrow', className)
  }, elementProps));
};
exports.Arrow = Arrow;