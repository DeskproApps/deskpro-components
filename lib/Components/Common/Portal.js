'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _noop = require('../utils/noop');

var _noop2 = _interopRequireDefault(_noop);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Portal mounts its children in the document body.
 */
var Portal = (_temp = _class = function (_React$Component) {
  _inherits(Portal, _React$Component);

  function Portal() {
    _classCallCheck(this, Portal);

    return _possibleConstructorReturn(this, (Portal.__proto__ || Object.getPrototypeOf(Portal)).apply(this, arguments));
  }

  _createClass(Portal, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.popup = document.createElement("div");
      document.body.appendChild(this.popup);
      this._renderLayer();
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      this._renderLayer();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      _reactDom2.default.unmountComponentAtNode(this.popup);
      document.body.removeChild(this.popup);
      this.popup = null;
    }
  }, {
    key: '_renderLayer',
    value: function _renderLayer() {
      var _this2 = this;

      _reactDom2.default.render(this.props.children, this.popup, function () {
        _this2.props.onUpdate(_this2);
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement('div', null);
    }
  }]);

  return Portal;
}(_react2.default.Component), _class.propTypes = {
  /**
   * Called when the child elements are mounted.
   */
  onUpdate: _propTypes2.default.func
}, _class.defaultProps = {
  onUpdate: _noop2.default
}, _temp);
exports.default = Portal;