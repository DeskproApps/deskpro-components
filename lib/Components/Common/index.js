'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListElement = exports.List = undefined;

var _List = require('./List');

var _ListElement = require('./ListElement');

exports.List = _List.List;
exports.ListElement = _ListElement.ListElement;