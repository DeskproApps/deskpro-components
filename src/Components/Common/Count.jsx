import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { numberFormat, THOUSANDTHS_SEPARATOR } from 'Components/utils/numbers';

/**
 * Renders a number as an inline element with thousandths formatting.
 */
const Count = ({ separator, children, className, ...props}) => (
  <span className={classNames('dp-count', className)} {...props}>
    {numberFormat(children, separator)}
  </span>
);

Count.propTypes = {
  /**
   * Character to use as the thousandths separator.
   */
  separator: PropTypes.string
};

Count.defaultProps = {
  separator: THOUSANDTHS_SEPARATOR
};

export default Count;
