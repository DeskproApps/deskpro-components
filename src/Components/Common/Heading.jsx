import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from 'Components/Icon';

/**
 * A header element which can be displayed with an icon and count number.
 */
const Heading = ({size, icon, count, children, className, ...props}) => {
  children = React.Children.toArray(children);

  if (icon !== undefined) {
    children.unshift(
      (typeof icon === 'string')
        ? <Icon key="icon" name={icon} />
        : React.cloneElement(
          icon,
          {key: "icon"}
        )
    );
  }
  if (count !== undefined) {
    children.push(
      <span key="count" className="dp-heading__count">
        {count}
      </span>
    );
  }

  props.className = classNames('dp-heading', className);
  return React.createElement(`h${size}`, props, children);
};

Heading.propTypes = {
  /**
   * A value from 1 to 6 indicating the type of header to use, e.g. h1, h2, h3, etc.
   */
  size: PropTypes.number,
  /**
   * Icon to display to the left of the text.
   */
  icon: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  /**
   * Number value to display to the right of the text.
   */
  count: PropTypes.number,
  /**
   * CSS classes to apply to the element.
   */
  className: PropTypes.string
};

Heading.defaultProps = {
  size: 1
};

export default Heading;
