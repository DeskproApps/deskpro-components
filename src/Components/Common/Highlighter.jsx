import React from 'react';
import PropTypes from 'prop-types';
import { stringHighlight } from 'utils/strings';

/**
 * Wraps a phrase in <i> tags for the sake of highlighting the phrase.
 */
const Highlighter = ({ highlight, element, children, ...props }) => (
  React.createElement('span', {
    ...props,
    dangerouslySetInnerHTML: { __html: stringHighlight(children, highlight, element) }
  })
);

Highlighter.propTypes = {
  /**
   * The word or phrase to highlight.
   */
  highlight: PropTypes.string.isRequired,
  /**
   * Type of element that surrounds the highlighted word.
   */
  element:   PropTypes.string,
  /**
   * The component children.
   */
  children:  PropTypes.string
};

Highlighter.defaultProps = {
  element: 'i'
};

export default Highlighter;
