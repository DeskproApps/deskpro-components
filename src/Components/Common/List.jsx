import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class List extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    children:  PropTypes.node,
  };

  static defaultProps = {
    className: '',
    children:  ''
  };

  render() {
    const {
      children, className, innerRef, ...elementProps
    } = this.props;
    return (
      <ul
        className={classNames('dp-list', className)}
        ref={innerRef}
        {...elementProps}
      >
        {children}
      </ul>
    );
  }
}
export default React.forwardRef((props, ref) => <List innerRef={ref} {...props} />);
