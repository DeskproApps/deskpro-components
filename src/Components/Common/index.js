import List from './List';
import ListElement from './ListElement';
import ListElementGroup from './ListElementGroup';
import Popper from './Popper';
import PopperIcon from './PopperIcon';
import Portal from './Portal';
import ToggleableList from './ToggleableList';
import QueryableList from './QueryableList';
import Heading from './Heading';
import Count from './Count';
import Subheading from './Subheading';
import Scrollbar from './Scrollbar';
import SelectableList from './SelectableList';

export {
  List,
  ListElement,
  ListElementGroup,
  Popper,
  PopperIcon,
  Portal,
  ToggleableList,
  QueryableList,
  Heading,
  Count,
  Subheading,
  Scrollbar,
  SelectableList,
};
