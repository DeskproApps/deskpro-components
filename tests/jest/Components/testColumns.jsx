import React from 'react';
import renderer from 'react-test-renderer';
import { Heading, Count } from '../../../src/Components/Common/index';
import {
  Column,
  Drawer,
  DrawerList,
  Item,
  ItemFilter,
  ItemList,
  ItemPopper,
  ItemSettings
} from '../../../src/Components/Columns/index';

const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

// https://github.com/facebook/react/issues/7740#issuecomment-247335106
const createNodeMock = () => ({
  addEventListener: () => {}
});

describe('>>> Columns --- Snapshot', () => {
  it('+++capturing Snapshot of Column', () => {
    const renderedValue = renderer.create(<Column className="test-class">
      <Heading>Test</Heading>
    </Column>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Drawer', () => {
    const renderedValue = renderer.create(<Drawer className="test-class">
      <Heading>
          Test
      </Heading>
      <ItemList>
        <Item>
            Item 1
          <Count>1</Count>
        </Item>
        <Item>
            Item 2
          <Count>0</Count>
        </Item>
      </ItemList>
    </Drawer>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of DrawerList', () => {
    const renderedValue = renderer.create(<DrawerList className="test-class">
      <Drawer className="test-class" />
    </DrawerList>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of ItemFilter', () => {
    const renderedValue = renderer.create(
      <ItemFilter className="test-class" />,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of ItemPopper', () => {
    const renderedValue = renderer.create(
      <ItemPopper className="test-class" opened />,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of ItemSettings', () => {
    const renderedValue = renderer.create(
      <ItemSettings className="test-class" opened />,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
