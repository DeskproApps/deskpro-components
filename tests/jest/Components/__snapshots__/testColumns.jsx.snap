// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Columns --- Snapshot +++capturing Snapshot of Column 1`] = `
<div
  className="dp-column test-class"
  role="tree"
>
  <h1
    className="dp-heading"
  >
    Test
  </h1>
</div>
`;

exports[`>>> Columns --- Snapshot +++capturing Snapshot of Drawer 1`] = `
<li
  aria-expanded={true}
  className="dp-list__element dp-column-drawer test-class"
  id="dp-column-drawer-1"
  role="treeitem"
>
  <h1
    aria-controls="dp-column-drawer-body-1"
    aria-label="Click to close."
    className="dp-heading"
    onClick={[Function]}
  >
    Test
    <i
      aria-hidden={true}
      className="fa fa-caret-up dp-icon dp-icon--s dp-column-drawer__arrow"
      onClick={[Function]}
      title="Close"
    />
  </h1>
  <div
    aria-hidden={false}
    className="dp-column-drawer__body dp-column-drawer__body--opened"
    id="dp-column-drawer-body-1"
  >
    <ul
      className="dp-list dp-drawer-item-list"
    >
      <li
        className="dp-list__element dp-drawer-item"
        data-dp-toggle-id={0}
        onClick={[Function]}
        onSelect={[Function]}
      >
        <span
          className="dp-drawer-item__pos-left"
        />
        <span
          className="dp-drawer-item__pos-middle"
        >
          Item 1
        </span>
        <span
          className="dp-drawer-item__pos-right"
        >
          <span
            className="dp-count"
          >
            1
          </span>
        </span>
      </li>
      <li
        className="dp-list__element dp-drawer-item"
        data-dp-toggle-id={1}
        onClick={[Function]}
        onSelect={[Function]}
      >
        <span
          className="dp-drawer-item__pos-left"
        />
        <span
          className="dp-drawer-item__pos-middle"
        >
          Item 2
        </span>
        <span
          className="dp-drawer-item__pos-right"
        >
          <span
            className="dp-count"
          >
            0
          </span>
        </span>
      </li>
    </ul>
  </div>
</li>
`;

exports[`>>> Columns --- Snapshot +++capturing Snapshot of DrawerList 1`] = `
<ul
  className="dp-list dp-column-drawer-list test-class"
  role="tree"
>
  <li
    aria-expanded={true}
    className="dp-list__element dp-column-drawer test-class"
    id="dp-column-drawer-2"
    role="treeitem"
  >
    <h1
      aria-controls="dp-column-drawer-body-2"
      aria-label="Click to close."
      className="dp-heading"
      onClick={[Function]}
    >
      Â 
      <i
        aria-hidden={true}
        className="fa fa-caret-up dp-icon dp-icon--s dp-column-drawer__arrow"
        onClick={[Function]}
        title="Close"
      />
    </h1>
    <div
      aria-hidden={false}
      className="dp-column-drawer__body dp-column-drawer__body--opened"
      id="dp-column-drawer-body-2"
    >
      
    </div>
  </li>
</ul>
`;

exports[`>>> Columns --- Snapshot +++capturing Snapshot of ItemFilter 1`] = `
<span
  className="test-class"
>
  <i
    className="fa fa-filter dp-icon dp-icon--s dp-popper-icon dp-icon__filter"
    onClick={[Function]}
  />
  <div />
</span>
`;

exports[`>>> Columns --- Snapshot +++capturing Snapshot of ItemPopper 1`] = `
<span
  className="test-class"
  opened={true}
>
  <div />
</span>
`;

exports[`>>> Columns --- Snapshot +++capturing Snapshot of ItemSettings 1`] = `
<span
  className="test-class"
>
  <i
    className="fa fa-cog dp-icon dp-icon--s dp-popper-icon dp-icon__settings"
    onClick={[Function]}
  />
  <div />
</span>
`;
